File: libs/shared/src/interfaces/login-user.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty, MinLength } from 'class-validator';

export class LoginUserDto {
  @IsNotEmpty()
  @ApiProperty({
    description: 'The username or email of the user',
    type: String,
    example: 'johndoe',
  })
  readonly usernameOrEmail!: string;
  @IsNotEmpty()
  @MinLength(6)
  @ApiProperty({
    description: 'The password of the user',
    type: String,
    example: '123456',
  })
  readonly password!: string;
}

File: frontend/lib/business_loc/login_user_dto.dart
class LoginUserDto {
  String usernameOrEmail, password;

  LoginUserDto({
    required this.usernameOrEmail,
    required this.password,
  });

  LoginUserDto.init()
      : usernameOrEmail = '',
        password = '';

  Map<String, dynamic> get toJson => {
        'usernameOrEmail': usernameOrEmail,
        'password': password,
      };
}